packages/java/src/components/Annotation.tsx

```diff
@@ -0,0 +1,20 @@
+import { code, mapJoin, Refkey, refkey } from "@alloy-js/core";
+import { Reference } from "./Reference.js";
+import { Child, Children } from "@alloy-js/core/jsx-runtime";
+import { collectNamedArguments } from "../arguments.js";
+
+export interface AnnotationProps {
+  type: Child;
+  value?: Child; // If one argument, provide this
+  arguments?: Record<string, Child>; // Arg name to value, provide if many named arguments
```

Comment:
arguments doesn't feel descriptive to me. Maybe `values` would be better? Ideally this could be `value?: Child | Record<string, Child>` but some objects can be children (refkeys, refs), so that check is a little risky I guess.

Reply:
I think I'll implement it as just `value?: Record<string, Child>` and if there is only one key just pass the value of that key. In Java any argument going to an annotation is named anyway, just if there's only one argument you don't have to specify the name.